packages/data-factory/lib/BlankNodeBindingsScoped.ts

```diff
@@ -0,0 +1,20 @@
+import type * as RDF from 'rdf-js';
+
+/**
+ * A blank node that is scoped to a set of bindings.
+ */
+export class BlankNodeBindingsScoped implements RDF.BlankNode {
+  public readonly termType: 'BlankNode' = 'BlankNode';
+  public readonly singleBindingsScope: true = true;
+  public readonly value: string;
+
+  public constructor(value: string) {
+    this.value = value;
+  }
+
+  public equals(other: RDF.Term | null | undefined): boolean {
+    // TODO: should we always return false given the nature of this class?
```

Comment:
I would say no. I think the current implementation is fine.
Even if we always return false here, invoking `equals` from another bnode would still return true, which could cause more problems than we would fix.

Reply:
Done!